databuilder/backends/databricks.py

```diff
@@ -7,6 +9,7 @@ class DatabricksBackend(BaseBackend):
 
     query_engine_class = SparkQueryEngine
     patient_join_column = "patient_id"
+    implements = [databuilder.tables.databricks]
```

Comment:
This means that the backend implementer has to jigger around in two different places and line things up across them. Did you consider putting the authoritative list here and dynamically generating the modules in `databuilder.tables` instead?

Reply:
Well the short answer is that we can't dynamically generate the stuff in `databuilder.tables` if we want autocomplete to work. But I'm also hopeful (perhaps naively so) that there won't be too much juggling across two places. If the implementer is writing a new set of ehrQL tables to reflect what's in their database then they may not need anything beyond the `DefaultBackend` which just maps the ehrQL schema directly to the database one. And if the implementer is trying to map an existing schema to their database then most of the work will be on the backend class, with the ehrQL schema as a fixed target.

Of course, I expect the reality to be a bit more messy than that. But I'm hopeful that with a bit of tooling and some extra options for backend classes we can make life sufficiently easy for backend implementers.

Reply:
Right.

I'd certainly prefer to make life easier for dataset definition authors than for backend implementers.